### ISSUE TITLE
Nintendon't needs to try harder to keep hard drives awake (also, why the fuck are memcards ever stored on hard drives)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [zabujard/nintendon-t#102]
```
What steps will reproduce the problem?
1. Use a hard drive that has the unfortunate tendency to sleep after five 
minutes of idle.
2. Play a game that does disc reads rarely enough to cause that.
3. Watch it run into a Disc Read Error next time it tries to read (or, god 
forbid, do memcard things).

What is the expected output? What do you see instead?
-> Nintendon't should a) either keep the memcard on the SD card or buffer 
writes to it so that they don't time out and fail from the game's point of 
view, and b) keep hard drives vigorously awake by sending occasional fake reads 
or something.

What revision of Nintendont are you using? On what system Wii/Wii U?
-> 2.167, Wii U. Hard drive is a bare 1 TB WD Green (WD10EADS) plugged into a 
Plugable USB3 toaster-lookin' docking station (USB-SATA-U3). AFAIK, there's no 
way to force either of these things to stay awake.

Please provide any additional information below.
-> Animal Crossing and the N64 Zelda ports are currently impossible to save in 
after a while on hard drives like these since they read the entire game from 
the disc on boot and then avoid disc reads entirely, which is why I'm of the 
opinion that memcards shouldn't be stored on the same device as the disc.
```

Original issue reported on code.google.com by `pauldacheez` on 8 Oct 2014 at 8:28

**Comment by GoogleCodeExporter** [comment]
```
Shouldn't you be able to disable sleep mode via WD SmartWare?
http://support.wdc.com/product/download.asp?modelno=WD10EADS&x=6&y=8

If you don't manage to to change the sleep mode, use another HDD.
```

Original comment by `barcaped...@gmail.com` on 10 Oct 2014 at 7:34

**Comment by GoogleCodeExporter** [comment]
```
WD Smartware won't do anything for a bare internal drive plugged into a 
third-party adapter. Heck, it doesn't even detect my iMac's internal WD disk. 
The Plugable adapter, on the other hand, does have a firmware update tool that 
can set the time-to-idle-sleep, but it's a Windows-only tool (as always) and I 
don't even know if I still have a Windows VM lying around.

Regardless if I can solve this issue myself, I want at least one of these two 
things done – a ton of Nintendon't users won't ever figure this out and will 
wrongly blame it for what their hard drive is doing.
```

Original comment by `pauldacheez` on 11 Oct 2014 at 4:11

**Comment by GoogleCodeExporter** [comment]
```
The same thing happens with me, With Super Mario Sunshine on current latest 
nintendont version(2.191). I disabled my external HDD standby feature (Samsung 
M3 portable "HX-M101TCB/G") using SDParm with success but the issue still 
happens, maybe the drive still goes into standby mode.
I can't open the enclosure and try to use the hdparm because it's logical board 
use USB 3.0 directly. 
It is possible to reproduce the problem earlier removing the usb cable from 
console, at next read the game stops on DRE screen.

Is it possible to make nintendont make a read from disk every 2 minutes?
```

Original comment by `machellos2` on 25 Oct 2014 at 3:26

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
At least for the Samsung M3 Portable External Hard Drives I found the solution.
This application enabled me to change successfully the disk sleep time.
Now the problem will happen here only after 2 hours instead of 5 minutes.

A fake write/read would be a painless solution, if possible.

Maybe it worth a shot for other Samsung external Hard drives.

http://www.seagate.com/support/downloads/item/samsung-drive-manager-master-dl/
```

Original comment by `machellos2` on 28 Oct 2014 at 1:58

**Comment by GoogleCodeExporter** [comment]
```
>mfw diosmios already has code for this
https://code.google.com/p/diosmios/source/browse/trunk/dip.c#370
```

Original comment by `pauldacheez` on 11 Nov 2014 at 12:12

**Comment by GoogleCodeExporter** [comment]
```
Fixed in r215, feel free to close. <3
```

Original comment by `pauldacheez` on 12 Nov 2014 at 6:51

**Comment by GoogleCodeExporter** [comment]
Original comment by `Howard_M...@yahoo.com` on 12 Nov 2014 at 9:01
- Changed state: **Verified**
